restapi/src/main/java/io/stargate/web/docsapi/service/query/condition/impl/ExistsCondition.java

```diff
@@ -70,13 +56,18 @@ public FilterOperationCode getFilterOperationCode() {
   /** {@inheritDoc} */
   @Override
   public boolean isEvaluateOnMissingFields() {
-    return false;
+    return !getQueryValue();
   }
 
   /** {@inheritDoc} */
   @Override
   public boolean test(Row row) {
-    // always test true when row is there :)
-    return true;
+    // if row exists then the test is true if query value is true
```

Comment:
could you elaborate more what's the use case for this "convoluted" logic?

Reply:
This is just to be "conceptually correct". We will never call `test(null)`, however, I wanted to ensure that the logic behind the condition is mapped here.

But I can also revert this and assume row is never null (as we actually do in the other conditions). WDYT?

Reply:
shouldn't we just return false in case `row == null`?

Reply:
I guess `row` is never `null` here, except when test code calls it.

However, a `null` row means the row _does not exist_ so if the condition is `$exists: false`, it makes sense to return `true`, I think.

Reply:
Since all the other conditions assume `Row` is not null, I will apply the same logic here..